contents/data-structure/code/Tree/Tree.ts

```diff
@@ -0,0 +1,64 @@
+class TreeNode {
+  parent: TreeNode | null;
+
+  children: TreeNode[];
+
+  data: any;
+
+  level: number;
+
+  constructor(key: any, level: number, parent: TreeNode | null) {
+    this.parent = parent;
+    this.data = key;
+    this.children = [];
+    this.level = level;
+  }
+
+  addChlid(key: any): void {
+    this.children.push(new TreeNode(key, this.level + 1, this));
+  }
+
+  printChildNodes(): void {
+    this.children.forEach((e) => console.log(e.data));
+  }
+
+  getNode(key: any): TreeNode | null {
+    if (key === this.data) return this;
+    if (this.children.length === 0) return null;
+
+    for (const child of this.children) {
+      const res = child.getNode(key);
+      if (res) return res;
+    }
+
+    return null;
+  }
+
+  removeNode(key: any): void {
+    const target = this.getNode(key);
+    const targetedChildren = target.parent.children;
+    targetedChildren.splice(
+      targetedChildren.findIndex((child) => child.data === key),
+      1
+    );
```

Comment:
```suggestion
    const targetedChildren = target.parent.children;
    target.parent.children = targetedChildren.filter((child) => child.data !== key);
```
이렇게 써도 잘 돌아가는지는 모르겠지만, 이렇게 쓰면 좀 더 간결하지 않을까요?

Reply:
제가 왜 저렇게 짰을까요?
훨씬 가독성 좋다고 생각합니다. 수정할게요!